solr/core/src/java/org/apache/solr/schema/FieldType.java

```diff
@@ -905,6 +905,7 @@ protected void checkSupportsDocValues() {
   protected static final String ENABLE_GRAPH_QUERIES = "enableGraphQueries";
   private static final String ARGS = "args";
   private static final String POSITION_INCREMENT_GAP = "positionIncrementGap";
+  protected static final String SCORE_OVERLAPS = "scoreOverlaps";
```

Comment:
Perhaps this ought to be a new parameter instead so that it's easier to toggle?  I suspect you've thought of this already and I'm curious about your rationale.

Reply:
I have been thinking a lot about this! 

- Solr currently exposes per-field query configuration as a fieldType param, not query time (see [autoGeneratePhraseQueries and enableGraphQueries](https://lucene.apache.org/solr/guide/6_6/field-type-definitions-and-properties.html#general-properties).
- Solr doesn't yet have a way to pass per-field configuration at query time (my email about multiple analyzers proposes one system for doing this)

To do the latter, ideally you'd have an API that could let you see multiple views/configs on the same field, such as the following which would search two query-time versions of the actor field

`q=action movies&qf=actor_syn actor_nosyn^10 title text&defType=edismax&qf.actor_nosyn.field=actor&qf.actor_nosyn.analyzer=without_synonyms&qf.actor_syn.field=actor&qf.actor_syn.analyzer=with_synonyms&qf.actor_syn&scoreOverlaps=pick_best`

I think this sort of syntax could be extremely powerful, and deal with the ability to configure multiple query time analyzers. But a bridge too far for this PR...

Reply:
I need to correct you one one point: Solr has had a syntax for per-field query parameters for a long time.  The syntax is `f.fieldName.parameterName`  e.g. `f.title.hl.snippets`   SolrJ's SolrParams has convenience methods for this on the implementation side. Perhaps you overlooked this because most users only use it in the context of faceting parameters, even though it's certainly not unique to faceting (as in the example above for highlighting).  I'm not aware of any query parser that uses it yet but they certainly could.

Any way, I suppose even if we agree we'd like some query time customizability of this (and other settings), it would still be nice to establish a default fallback on the FieldType.